internal/controller/validation/image_validator.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2024 Apache Software Foundation (ASF)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"archive/tar"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/serverlessworkflow/sdk-go/v2/model"
+
+	operatorapi "github.com/apache/incubator-kie-kogito-serverless-operator/api/v1alpha08"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type imageValidator struct{}
+
+var workflowPathRegex = regexp.MustCompile(`^deployments/app/[^/]+\.sw\.(json|yaml)$`)
+
+func (v *imageValidator) Validate(ctx context.Context, client client.Client, sonataflow *operatorapi.SonataFlow, req ctrl.Request) error {
+	equals, err := validateImage(ctx, sonataflow)
+	if err != nil {
+		return err
+	}
+	if !equals {
+		return fmt.Errorf("Workflow, defined in the image %s doesn't match deployment workflow", sonataflow.Spec.PodTemplate.Container.Image)
+	}
+	return nil
+}
+
+func NewImageValidator() Validator {
+	return &imageValidator{}
+}
+
+func validateImage(ctx context.Context, sonataflow *operatorapi.SonataFlow) (bool, error) {
+	isInKindRegistry, _, err := imageStoredInKindRegistry(ctx, sonataflow.Spec.PodTemplate.Container.Image)
```

Comment:
@wmedvede As I can see, we use kind-registry and `sonataflow.Spec.PodTemplate.Container.Image` only in the tests, where `data.localRegistryHosting.v1.hostFromClusterNetwork` set. In case we want to use the same scenario with the minikube, it's better to add to `install-minikube.sh` registry and ConfigMap from the `create-kind-cluster-with-registry.sh`.